view/theme/frio/js/theme.js

```diff
@@ -764,12 +764,108 @@ function htmlToText(htmlString) {
  * @param {boolean} un    Whether to perform an activity removal instead of creation
  */
 function doActivityItemAction(ident, verb, un) {
-	if (verb.indexOf("attend") === 0) {
-		$(".item-" + ident + " .button-event:not(#" + verb + "-" + ident + ")").removeClass("active");
+	_verb = un ? 'un' + verb : verb;
+	var thumbsClass = '';
+	switch (verb) {
+		case 'like':
+			thumbsClass = 'fa-thumbs-up';
+			break;
+		case 'dislike':
+			thumbsClass = 'fa-thumbs-down';
+			break;
+		case 'announce':
+			thumbsClass = 'fa-retweet';
+	}
+	// remindert to remove the like-rotator from templates
+	//$('#like-rotator-' + ident.toString()).show();
+	if (verb.indexOf('announce') === 0 ) {
+		// Share-Button(s)
+		// remove share-symbol, to replace it by rotator
+		$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').removeClass('fa-share');
+		$('button[id^=announce-' + ident.toString() + '] i:first-child').removeClass('fa-retweet');
+		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
+		if ($('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').length == 0) {
+			// append rotator to the shareMenu-button for small media
+			$('<img id="waitfor-' + verb + '-' + ident.toString() + '">')
+				.attr('src', 'images/rotator.gif')
+				.addClass('fa')
+				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' )).show();
+		} else {
+			$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').show()
```

Comment:
Please use the [`.attr(attributes)`](https://api.jquery.com/attr/#attr-attributes) form to set the `id` attribute along the `src` one and avoid writing literal HTML as much as possible
```suggestion
			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```